platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/proof/tree/StateProofPayload.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.merkle.proof.tree;
+
+import com.swirlds.common.crypto.Cryptography;
+import com.swirlds.common.io.SelfSerializable;
+import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.swirlds.common.io.streams.SerializableDataOutputStream;
+import com.swirlds.common.merkle.MerkleLeaf;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.security.MessageDigest;
+
+/**
+ * A node in a state proof tree containing a payload.
```

Comment:
This could benefit from a slightly expanded description, e.g. "A payload is a merkle leaf whose contents will be proven by the state proof" or something like that

Reply:
```
/**
 * A node in a state proof tree containing a payload. Each payload corresponds to a merkle leaf in the original
 * state that is being proven by the state proof.
 */
```